core/src/main/java/org/kiwix/kiwixmobile/core/page/history/viewmodel/HistoryState.kt

```diff
@@ -29,24 +30,16 @@ data class HistoryState(
   override val showAll: Boolean,
   override val currentZimId: String?,
   override val searchTerm: String = ""
-) : PageState {
-  val isInSelectionState = pageItems.any(HistoryItem::isSelected)
-
-  val historyListItems: List<HistoryListItem> =
+) : PageState() {
+  override val filteredPageItems: List<HistoryListItem> =
     HeaderizableList<HistoryListItem, HistoryItem, DateItem>(pageItems
       .filter { showAll || it.zimId == currentZimId }
```

Comment:
this filtering logic is duplicated in your states, I think it makes sense to perhaps rename this `visiblePageItems` and add a default implementation to the interface for `filteredItems`